### PR TITLE
[cmd/opampsupervisor] Add configurable supervisor logging

### LINKED ISSUE #35466: [cmd/opampsupervisor] Add configurable supervisor logging
### Component(s)

cmd/opampsupervisor

### Is your feature request related to a problem? Please describe.

Right now the supervisor strictly logs to stdout which can be a pain point when trying to monitor the supervisor's logs. 

### Describe the solution you'd like

Add the option to configure how the supervisor logs data like log level and output destinations.

### Describe alternatives you've considered

You could just pipe the process output to a file, but that seems tedious. That approach also doesn't allow configuring log level output or other helpful logging configuration options.

### Additional context

I have a solution in progress for this feature request.

### PR DESCRIPTION
**Pull request by dpaasman00**
**Description:** <Describe what has changed.>

Adds configurable logging for the supervisor. Lays ground work to expand with more config options and to configure other telemetry signals. Meant to mimic how collector telemetry/logging is configured.

**Link to tracking Issue:** <Issue number if applicable> Closes #35466 

**Testing:** <Describe what testing was performed and which tests were added.>
Tested running the supervisor with default and different log levels and outputs specified.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by djaglowski — APPROVED** [review]

**Review by evan-bradley — COMMENTED** [review]
Thanks for adding this.

File: cmd/opampsupervisor/supervisor/config/config.go [hunk 1]
```diff
@@ -150,6 +155,16 @@ type AgentDescription struct {
 	NonIdentifyingAttributes map[string]string `mapstructure:"non_identifying_attributes"`
 }
 
+type Telemetry struct {
+	// TODO: flesh out other configurable telemetry options
```
[the remark this answers is not in the document]
Comment (dpaasman00) [reply cmd/opampsupervisor/supervisor/config/config.go:159]: I can add definitely make an issue for this.

[the remark this answers is not in the document]
Comment (dpaasman00) [reply cmd/opampsupervisor/supervisor/config/config.go]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35582

[the remark this answers is not in the document]
Comment (evan-bradley) [reply cmd/opampsupervisor/supervisor/config/config.go]: Really appreciate it, thanks.

**Review by evan-bradley — APPROVED** [review]
LGTM, thanks.

**Pull request merged**